### PR TITLE
chore: refactor phone-input unit tests to use Jest assertions.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Proposed Changes

Similar to https://github.com/Automattic/wp-calypso/pull/64743 and https://github.com/Automattic/wp-calypso/pull/64753, this PR updates the unit tests in `client/components/phone-input` to use Jest assertions.

This will remove more than 70 errors from the Check code style build.

image

Key changes:
- split long test file into several smaller chunks;
- use Jest assertions instead;
- use Jest parametrization instead.

#### Testing Instructions

Ensure the following:
  - [ ] Calypso E2E (desktop)
  - [ ] Calypso E2E (mobile)
  - [ ] Pre-Release E2E
  - [ ] Code Style
  - [ ] Check code style
  - [ ] Unit Tests

Related to https://github.com/Automattic/wp-calypso/pull/64753

### CODE REVIEW

File: client/components/phone-input/test/phone-number__format-number.test.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+describe( 'National Format', () => {
+	test.each( [
+		{ number: '4252222222', country: 'US', expected: '(425) 222-2222' },
+		{ number: '05325555555', country: 'TR', expected: '0532 555 55 55' },
+		{ number: '0215369851', country: 'AU', expected: '02 1536 9851' },
+		{ number: '3911711711', country: 'IT', expected: '391 171 1711' },
+		{ number: '5512345678', country: 'BR', expected: '055 1234-5678' },
+		{ number: '9100123456', country: 'IN', expected: '091001 23456' },
+		{ number: '00000123456', country: 'UK', expected: '01234 56' },
+	] )( `Format full length numbers`, function ( { number, country, expected } ) {
```
Comment (dpasque) [inline client/components/phone-input/test/phone-number__format-number.test.js]: Do we want to add some part of the input to the test name here so the tests have different names? Right now they all have the same name, which might make debugging a single failure a pinch slower.

File: client/components/phone-input/test/phone-number__format-number.test.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		expect( result ).toEqual( expected );
+	} );
+} );
+
+describe( 'Sanitization', () => {
+	test.each( [
+		{ number: '1aaaa', country: 'US', expected: '1' },
+		{ number: '1a', country: 'US', expected: '1' },
+	] )(
+		'Strip non-digits on string of length less than 3',
```
Comment (dpasque) [inline client/components/phone-input/test/phone-number__format-number.test.js]: Same here -- should we add something from the input to the test name?

File: client/components/phone-input/test/phone-number__misc.test.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		}
+	);
+} );
+
+describe( 'indexOfStrictSubsequenceEnd', function () {
+	test.each( [
+		{ array1: [ 'a', 'b' ], array2: [ 'a', 'b', 'c' ], expected: 2 },
+		{ array1: [ 'a', 'b' ], array2: [ 'a', 'd', 'b', 'c' ], expected: 3 },
+	] )(
+		`Correct index for subsequence is found for inputs $array1, array2`,
```
Comment (dpasque) [inline client/components/phone-input/test/phone-number__misc.test.ts]: I think you're missing a "$" here! `$array2`

File: client/components/phone-input/test/phone-number__misc.test.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		{ oldvalue: '(802) 687-6234', newvalue: '(802) 555-6234', position: 6, expected: 9 },
+		{ oldvalue: '124', newvalue: '1234', position: 2, expected: 3 },
+		{ oldvalue: '124', newvalue: '123-4', position: 2, expected: 3 },
+		{ oldvalue: '224', newvalue: '223-764', position: 2, expected: 6 },
+		{ oldvalue: '124-4', newvalue: '124-64', position: 4, expected: 5 },
+		{ oldvalue: '(802) 614-21', newvalue: '(802) 619-421', position: 8, expected: 9 },
+		{ oldvalue: '(802) 222-2222', newvalue: '6', position: 14, expected: 1 },
+		{ oldvalue: '+1 802 222-2222', newvalue: '6', position: 15, expected: 1 },
+	] )(
+		`Returns correct cursor position for inputs $oldvalue, $, newvalue`,
```
Comment (dpasque) [inline client/components/phone-input/test/phone-number__misc.test.ts]: I think something got messed up in the name here, I believe this should read...

`Returns correct cursor position for inputs $oldvalue, $newvalue`

**Review by dpasque — APPROVED** [review]
A few small nits re: test naming, but otherwise LGTM! 🚀

File: client/components/phone-input/test/phone-number__format-number.test.js (hunk 1 above)
Comment (worldomonation) [reply client/components/phone-input/test/phone-number__format-number.test.js]: Ah yes, that's my oversight! Thanks!

File: client/components/phone-input/test/phone-number__misc.test.ts (hunk 4 above)
Comment (worldomonation) [reply client/components/phone-input/test/phone-number__misc.test.ts]: I was using regex to mass-edit the parametrized inputs and I believe that's why it got messed up.

Comment (worldomonation) [reply client/components/phone-input/test/phone-number__misc.test.ts]: Addressed in e10751336012e40c886f517b08cc4f2cbe15d2e0.

File: client/components/phone-input/test/phone-number__format-number.test.js (hunk 1 above)
Comment (worldomonation) [reply client/components/phone-input/test/phone-number__format-number.test.js]: Addressed in e10751336012e40c886f517b08cc4f2cbe15d2e0.

File: client/components/phone-input/test/phone-number__format-number.test.js (hunk 2 above)
Comment (worldomonation) [reply client/components/phone-input/test/phone-number__format-number.test.js]: Fixed in e10751336012e40c886f517b08cc4f2cbe15d2e0.

File: client/components/phone-input/test/phone-number__misc.test.ts (hunk 3 above)
Comment (worldomonation) [reply client/components/phone-input/test/phone-number__misc.test.ts]: Fixed in e10751336012e40c886f517b08cc4f2cbe15d2e0.

**Pull request merged**